File: src/orchestration/Python/nutrition_agent.py
from __future__ import annotations

from agent import Agent
from orchestrator_result import OrchestratorResult


class NutritionAgent(Agent):
    @property
    def name(self) -> str:
        return "NutritionAgent"

    def can_handle(self, context) -> bool:
        return False

    def execute(self, context) -> OrchestratorResult:
        return OrchestratorResult("NutritionAgent is not active yet.", self.name, [], [], ["Assistant", "AssistantOrchestrator", self.name], 0.0, ["Agent is not active."], context.messages_to_persist)

File: src/orchestration/DotNet/NutritionAgent.cs
namespace Orchestration.DotNet;

public sealed class NutritionAgent : IAgent
{
    public string Name => "NutritionAgent";

    public bool CanHandle(OrchestratorContext context) => false;

    public Task<OrchestratorResult> ExecuteAsync(OrchestratorContext context, CancellationToken cancellationToken = default) =>
        Task.FromResult(new OrchestratorResult("NutritionAgent is not active yet.", Name, [], [], ["Assistant", "Orchestrator", Name], TimeSpan.Zero, ["Agent is not active."], context.MessagesToPersist));
}